src/XamlX/Ast/Clr.cs

```diff
@@ -68,17 +71,44 @@ class XamlAstClrProperty : XamlAstNode, IXamlAstPropertyReference
         }
 
         public XamlAstClrProperty(IXamlLineInfo lineInfo, string name, IXamlType declaringType, 
+            TransformerConfiguration cfg,
             IXamlMethod getter, IEnumerable<IXamlPropertySetter> setters) : base(lineInfo)
         {
             Name = name;
             DeclaringType = declaringType;
             Getter = getter;
             if (setters != null)
+            {
                 Setters.AddRange(setters);
+            }
+
+            if (getter != null)
+            {
+                PropertyType = getter.ReturnType;
+            }
+            else
+            {
+                foreach (var setter in Setters)
```

Comment:
I'm not sure if we should mutate setters here. It also treats
```
public void set_Foo(int value);
```

as `IEnumerable<int>` property. Which doesn't really make sense and feels like a hack just for whitespace handling

Reply:
It should do that only if the setter was added with `BinderParameters.AllowMultiple`. Would that be the case for `set_Foo' ?

(I also didn't like all of this when I wrote it, to be honest)

Reply:
I think we need to add an extra marker to `Add`-like setters, so you could check for that in  ApplyWhitespaceNormalization transformer

Reply:
I thought that's what `BinderParameters.AllowMultiple` is for? 
I coded this under the assumption that add-style setters are marked `BinderParameters.AllowMultiple = true`, while set-style setters are not.